src/pharmpy/model/external/nonmem/update.py

```diff
@@ -1584,21 +1623,30 @@ def update_estimation(control_stream, model):
         new_parameter_uncertainty_method = est.parameter_uncertainty_method
 
     if old_parameter_uncertainty_method is None and new_parameter_uncertainty_method is not None:
-        # Add $COVARIANCE
+        # Add parameter uncertainty step
         last_est_rec = control_stream.get_records('ESTIMATION')[-1]
         idx_cov = control_stream.records.index(last_est_rec)
-        if new_parameter_uncertainty_method == 'SANDWICH':
-            covrec_ = '$COVARIANCE UNCONDITIONAL PRINT=E PRECOND=1'
-        elif new_parameter_uncertainty_method == 'CPG':
-            covrec_ = '$COVARIANCE MATRIX=S UNCONDITIONAL PRINT=E PRECOND=1'
-        elif new_parameter_uncertainty_method == 'OFIM':
-            covrec_ = '$COVARIANCE MATRIX=R UNCONDITIONAL PRINT=E PRECOND=1'
-        covrec = create_record(f'{covrec_}\n')
-        control_stream = control_stream.insert_record(covrec, at_index=idx_cov + 1)
+
+        if new_parameter_uncertainty_method == "EFIM":
+            control_stream = add_efim_records(control_stream, idx_cov, last_est_rec)
+        else:
+            control_stream = add_covariance_record(
+                control_stream, idx_cov, new_parameter_uncertainty_method
+            )
+
     elif old_parameter_uncertainty_method is not None and new_parameter_uncertainty_method is None:
-        # Remove $COVARIANCE
-        covrecs = control_stream.get_records('COVARIANCE')
-        control_stream = control_stream.remove_records(covrecs)
+        # If old_parameter_uncertainty_method is EFIM, remove EFIM records
+        if old_parameter_uncertainty_method == "EFIM":
+            record_types = ["DESIGN", "MSFI", "INPUT", "DATA", "PROBLEM"]
```

Comment:
Why is this in reverse order?

Reply:
Because $PROBLEM needs to be removed last, since we are using that to be sure we only remove records from the last $PROBLEM